indy-utils/src/main/kotlin/com/luxoft/blockchainlab/hyperledger/indy/helpers/PoolHelper.kt

```diff
@@ -0,0 +1,88 @@
+package com.luxoft.blockchainlab.hyperledger.indy.helpers
+
+import com.luxoft.blockchainlab.hyperledger.indy.utils.EnvironmentUtils
+import org.hyperledger.indy.sdk.pool.InvalidPoolException
+import org.hyperledger.indy.sdk.pool.Pool
+import org.hyperledger.indy.sdk.pool.PoolJSONParameters
+import org.hyperledger.indy.sdk.pool.PoolJSONParameters.OpenPoolLedgerJSONParameter
+import org.hyperledger.indy.sdk.pool.PoolLedgerConfigExistsException
+import java.io.File
+import java.io.IOException
+import java.nio.file.Files
+import java.util.concurrent.ExecutionException
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.util.*
+import kotlin.math.absoluteValue
+
+
+/**
+ * Helps to manage ledger pool
+ */
+object PoolHelper {
+
+    const val DEFAULT_POOL_NAME = "default_pool"
+
+    /**
+     * Creates (or not if exists) pool ledger files and returns [Pool] object
+     *
+     * If pool already opened, creates a symlink with random name that targets needed pool
+     *
+     * @param genesisFile: [File] - file with genesis transaction
+     * @param poolName: [String] - name of pool
+     * @param poolConfig: [OpenPoolLedgerJSONParameter] - pool config (when you create new pool)
+     * @return: [Pool] - target pool handle
+     */
+    fun getPool(
+        genesisFile: File,
+        poolName: String = DEFAULT_POOL_NAME,
+        poolConfig: OpenPoolLedgerJSONParameter = OpenPoolLedgerJSONParameter(null, null)
+    ): Pool {
+
+        val ledgerConfig = PoolJSONParameters.CreatePoolLedgerConfigJSONParameter(genesisFile.absolutePath)
+
+        try {
+            Pool.createPoolLedgerConfig(poolName, ledgerConfig.toJson()).get()
+        } catch (e: ExecutionException) {
+            if (e.cause !is PoolLedgerConfigExistsException) throw e
+        }
+
+        Pool.setProtocolVersion(2).get()
+
+        return try {
+            Pool.openPoolLedger(poolName, poolConfig.toJson()).get()
+        } catch (ex: ExecutionException) {
+            if (ex.cause !is InvalidPoolException) throw ex
+
+            val linkName = symlinkPool(poolName)
```

Comment:
The logic of symlinks should be moved away from production code. Currently the error is masked and this is wrong concept. If it requires for test then either wrapper need to be implemented or error handler. In production it may prevent to unexpected behavior. 

If multiple opened pools are supported by wrapper it should be contributed there.

Reply:
Ok, I'll move symlinkPool function to tests.